main.cpp

```diff
@@ -17,8 +17,8 @@ int main(int argc, char **argv)
         const char *filtersOptions {"begrs"};
 
         // Get filter flag
-        char filter {getopt(argc, argv, filtersOptions)};
-        
+        char filter {static_cast<char>(getopt(argc, argv, filtersOptions))};
```

Comment:
Why not `unsigned`?

Reply:
yes, It should be unsigned char rather than signed. I have updated it.

Reply:
My mistake. It should be `signed char` or simply `char` only because it can return `-1` if there are no more options to process.